### ISSUE TITLE
fre pp validate doesn't check for circular dependencies in large workflows

### ISSUE THREAD

**Issue by ilaflott** [NOAA-GFDL/fre-cli#106]
`fre pp validate` calls `cylc validate`, but without the `--check-circular` flag for avoiding circular task dependencies in a workflow. As a result, we get the following output:

```
(freCli4) @an102 [ESM4p2_piControl_... $] fre pp validate -e FOO_exp -p BAR_platform -t BAZ_target
[INFO] app/analysis/rose-app.conf
[INFO] app/combine-statics/rose-app.conf
[INFO] app/combine-timeavgs/rose-app.conf
[INFO] app/make-timeavgs/rose-app.conf
[INFO] app/make-timeseries/rose-app.conf
[INFO] app/mask-atmos-plevel/rose-app.conf
[INFO] app/regrid-xy/rose-app.conf
[INFO] app/remap-pp-components/rose-app.conf
[INFO] app/rename-split-to-pp/rose-app.conf
[INFO] app/split-netcdf-python/rose-app.conf
[INFO] app/split-netcdf/rose-app.conf
[INFO] rose-suite.conf
...
...
...
WARNING - Number of tasks is > 100; will not check graph for circular dependencies. To enforce this check, use the option --check-circular.
Valid for cylc-8.3.0
```

A successful solution to this issue adds the `--check-circular` flag the appropriate spot in `fre-cli`, and as a result, the above warning will not be shown upon successful workflow configuration.

**Comment by ilaflott** [comment]
still relevant. bump.

**Comment by ilaflott** [comment]
basically, take this line: https://github.com/NOAA-GFDL/fre-cli/blob/a9730e039274739e5dc8710d82dcc09578735d18/fre/pp/validate_script.py#L27

and add `--check-circular` condition flag into a `fre pp validate .`

**Comment by ceblanton** [comment]
@singhd789 , I think `fre pp validate` is also on the retirement block, to be folded into the `fre worklfow` tools. Do you agree?

Ian, we are planning to move the workflow parts of `fre pp` into `fre workflow` and have `fre pp` be more of a sub-tool toolset.

I think this is still a valid issue, though! Just to target `fre workflow validate` instead.

Thanks!

**Comment by singhd789** [comment]
> [@singhd789](https://github.com/singhd789) , I think `fre pp validate` is also on the retirement block, to be folded into the `fre worklfow` tools. Do you agree?
> 
> Ian, we are planning to move the workflow parts of `fre pp` into `fre workflow` and have `fre pp` be more of a sub-tool toolset.
> 
> I think this is still a valid issue, though! Just to target `fre workflow validate` instead.
> 
> Thanks!

I agree, any tools that pertain to cylc workflows that can be extended beyond the postprocessing workflow can be ingested into a `fre workflow` subtool! With a little edits for the `rose` stuff and `make_workflow_name`, maybe we can move almost this exact tool over to `fre workflow validate`

**Comment by singhd789** [comment]
However, I will say, that subtool doesn't exist yet (slated for 2026.02.alpha2), soooo if we did want to just include that capability real quick, the PR following this issue isn't big and could serve as a reminder to keep that `check-circular` option in when that subtool is developed
